src/server/gc_worker/compaction_runner.rs

```diff
@@ -371,13 +371,20 @@ impl<S: GcSafePointProvider, R: RegionInfoProvider + 'static, E: KvEngine>
         let start_time = Instant::now();
         let mut processed_count = 0;
         let total_candidates = candidates.len();
+        let check_interval = config.auto_compaction.check_interval.0;
         fail_point!("gc_worker_auto_compaction_start_compacting");
 
         for (index, candidate) in candidates.into_iter().enumerate() {
             if self.check_stopped() {
                 return None; // Stopped
             }
 
+            // Check if we've exceeded the check interval, return to start next round
+            if start_time.elapsed() >= check_interval {
```

Comment:
Could we skip already checked or compacted candidates in the next round if timeout ?

Reply:
I think it against the will. The reason of setting a time window it to avoid "starving" the hot regions. If a region is selected as candidate again in the next round, it means it is worthy to be compacted. What's your concern here?

Reply:
The repeated processing of duplicated or already-compacted candidates in `evaluate_range_candidate` may incur significant computational overhead.

Reply:
It may be. Howeverm I guess "check interval" can be interpreted as "how often we want to revaluate region compaction priority", if you found that, in some cases, "check" is too much of a overhead in "check then compact" logic, we can just extend "check interval". It will have the same behavior as "skip already checked or compacted candidates", right?